test/envtest/kongpluginbinding_managed_test.go

```diff
@@ -117,113 +72,326 @@ func TestKongPluginBindingManaged(t *testing.T) {
+		})
+		updateKongServiceStatusWithProgrammed(t, ctx, clientNamespaced, kongService, serviceID, cp.GetKonnectStatus().GetKonnectID())
+		kongRoute := deployKongRouteAttachedToService(t, ctx, clientNamespaced, kongService,
+			WithAnnotation(consts.PluginsAnnotationKey, rateLimitingkongPlugin.Name),
+		)
+		t.Cleanup(func() {
+			require.NoError(t, clientNamespaced.Delete(ctx, kongRoute))
+		})
+		updateKongRouteStatusWithProgrammed(t, ctx, clientNamespaced, kongRoute, routeID, cp.GetKonnectStatus().GetKonnectID(), serviceID)
+
+		t.Logf("waiting for KongPluginBinding to be created")
+		kongPluginBinding := watchFor(t, ctx, wKongPluginBinding, watch.Added,
+			func(kpb *configurationv1alpha1.KongPluginBinding) bool {
+				return kpb.Spec.Targets.RouteReference != nil &&
+					kpb.Spec.Targets.RouteReference.Name == kongRoute.Name &&
+					kpb.Spec.PluginReference.Name == rateLimitingkongPlugin.Name
+			},
+			"KongPluginBinding wasn't created",
+		)
+		t.Logf(
+			"checking that managed KongPlugin %s gets plugin-in-use finalizer added",
+			client.ObjectKeyFromObject(rateLimitingkongPlugin),
+		)
+		_ = watchFor(t, ctx, wKongPlugin, watch.Modified,
+			func(kp *configurationv1.KongPlugin) bool {
+				return kp.Name == rateLimitingkongPlugin.Name &&
+					controllerutil.ContainsFinalizer(kp, consts.PluginInUseFinalizer)
+			},
+			"KongPlugin wasn't updated to get plugin-in-use finalizer added",
+		)
+
+		t.Logf("delete managed kongPluginBinding %s, then check it gets recreated", client.ObjectKeyFromObject(kongPluginBinding))
+		require.NoError(t, clientNamespaced.Delete(ctx, kongPluginBinding))
+		kongPluginBinding = watchFor(t, ctx, wKongPluginBinding, watch.Added,
+			func(kpb *configurationv1alpha1.KongPluginBinding) bool {
+				return kpb.Spec.Targets.RouteReference != nil &&
+					kpb.Spec.Targets.RouteReference.Name == kongRoute.Name &&
+					kpb.Spec.PluginReference.Name == rateLimitingkongPlugin.Name
+			},
+			"KongPluginBinding wasn't recreated",
+		)
+
+		assert.EventuallyWithT(t, func(c *assert.CollectT) {
+			assert.True(c, sdk.PluginSDK.AssertExpectations(t))
+		}, waitTime, tickTime)
+
+		t.Logf(
+			"remove annotation from KongRoute %s and check that managed KongPluginBinding %s gets deleted, "+
+				"then check that gateway.konghq.com/plugin-in-use finalizer gets removed from %s KongPlugin",
+			client.ObjectKeyFromObject(kongRoute),
+			client.ObjectKeyFromObject(kongPluginBinding),
+			client.ObjectKeyFromObject(rateLimitingkongPlugin),
+		)
+
+		sdk.PluginSDK.EXPECT().
+			DeletePlugin(mock.Anything, cp.GetKonnectStatus().GetKonnectID(), mock.Anything).
+			Return(
+				&sdkkonnectops.DeletePluginResponse{
+					StatusCode: 200,
 				},
+				nil,
+			)
+
+		t.Logf(
+			"checking that managed KongPlugin %s gets plugin-in-use finalizer removed",
+			client.ObjectKeyFromObject(rateLimitingkongPlugin),
+		)
+		wKongPlugin = setupWatch[configurationv1.KongPluginList](t, ctx, clientWithWatch, client.InNamespace(ns.Name))
```

Comment:
Why not use the already-created watch?

Reply:
Makes sense. 62e4f38884f886001032baeeb6249398b104f1d4 removes it.